parquet/src/main/scala/magnolify/parquet/ParquetType.scala

```diff
@@ -85,19 +85,23 @@ sealed trait ParquetType[T] extends Serializable {
 object ParquetType {
   private val logger = LoggerFactory.getLogger(this.getClass)
 
-  implicit def apply[T](implicit f: ParquetField.Record[T], pa: ParquetArray): ParquetType[T] =
+  implicit def apply[T](implicit f: ParquetField[T], pa: ParquetArray): ParquetType[T] =
     ParquetType(CaseMapper.identity)
 
   def apply[T](
     cm: CaseMapper
-  )(implicit f: ParquetField.Record[T], pa: ParquetArray): ParquetType[T] =
-    new ParquetType[T] {
-      override def schema: MessageType = Schema.message(f.schema(cm))
-
-      override val avroCompat: Boolean = pa == ParquetArray.AvroCompat.avroCompat || f.hasAvroArray
-      override def write(c: RecordConsumer, v: T): Unit = f.write(c, v)(cm)
-      override def newConverter: TypeConverter[T] = f.newConverter
-    }
+  )(implicit f: ParquetField[T], pa: ParquetArray): ParquetType[T] = f match {
+    case r: ParquetField.Record[_] =>
```

Comment:
Was it possible to still have a Record as implicit parameter?

Reply:
yes, the `magnolia` macro was made to always return a `Record`